python/common/org/python/types/Float.java

```diff
@@ -169,7 +171,15 @@ public Float(double value) {
         __doc__ = ""
     )
     public org.python.Object __ne__(org.python.Object other) {
-        throw new org.python.exceptions.NotImplementedError("float.__ne__() has not been implemented.");
+        if (other instanceof org.python.types.Int || other instanceof org.python.types.Float ||
```

Comment:
I think this looks fine for the types you're including, but in my testing you _can_ compare a string to a float.

```
>>> float(1.0) != "1.0"
True
```

So I'm super curious why the string test passes for float, given your code. Is this a python 3.4 vs 3.5 difference? (I'm on 3.5)